modal.js

```diff
@@ -29,4 +29,30 @@ modalBtnNextPage.forEach((modalBtnNextPage) => {
       currentStep += 1;
     }
   });
-});
\ No newline at end of file
+});
+
+/* FETCH API BOOKING 1 */
+const dateField = document.querySelector(".modal__date");
+dateField.setAttribute("min", new Date().toISOString().split('T')[0]);
+let timeSlots;
+async function getAvailableTimes(date) {
+  const res = await fetch(
+    `https://lernia-sjj-assignments.vercel.app/api/booking/available-times?date=${date}&challenge=3`
+    // Todo: Ändra "challenge=3" efter att bokningsknappen lagts till på korten
+  );
+  const data = await res.json();
+  return data.slots;
+}
+dateField.addEventListener("change", (event) => {
+  if (event.target.value === "") {
+    searchBtn.disabled = true;
+  } else {
+    searchBtn.disabled = false;
+  }
+});
+const searchBtn = document.querySelector(".modal__searchBtn");
+searchBtn.disabled = true; // Knappen är disabled för att man inte ska kunna gå vidare till nästa steg utan att välja ett datum
+searchBtn.addEventListener("click", () => {
+  const date = dateField.value;
+  timeSlots = getAvailableTimes(date);
```

Comment:
Instead of hardcoding the ID 3 inside of the function, where it's "hidden", you could hardcode it (and move your `Todo` comment) here:

```js
getAvailableTimes(date, 3);
```